src/editor/grid.rs

```diff
@@ -29,13 +29,15 @@ pub struct CharacterGrid {
     pub height: usize,
 
     lines: Vec<GridLine>,
```

Comment:
Maybe add a comment to this file just sign posting that a ring buffer is being used. I could imagine somebody coming after and scratching their head a bit about whats going on here. Would be nice to point them in the right direction

Reply:
For extra points, you could even pull the lines and top_index fields out into their own lines_ring_bufffer struct which exposes a safe interface that ensures you interact with it properly. But up to you if thats too much overhead

Reply:
Not doing the api change here. Just added a comment above lines indicating that it is a ring buffer.